caffe2/CMakeLists.txt

```diff
@@ -363,6 +363,7 @@ if (NOT INTERN_BUILD_MOBILE OR NOT BUILD_CAFFE2_MOBILE)
     ${TORCH_SRC_DIR}/csrc/jit/pickler.cpp
     ${TORCH_SRC_DIR}/csrc/jit/unpickler.cpp
     ${TORCH_SRC_DIR}/csrc/jit/graph_executor.cpp
+    ${TORCH_SRC_DIR}/csrc/jit/backward.cpp
```

Comment:
When we add a new file, we also need to add it to `tools/build_variables.py` (that file generates build files for internal builds).

Also, a nit: the name `backward.cpp` seems to me too generic, maybe use something more concrete? `grad_executor` for instance, but I don't insist on that particular option.